lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -780,10 +782,15 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
     }
 
     markRowAsContextMenuSource(selectedRow: DataRow): void {
+        this.selectedRowId = selectedRow.id ? selectedRow.id : '';
         this.data.getRows().forEach((row) => row.isContextMenuSource = false);
         selectedRow.isContextMenuSource = true;
     }
 
+    setRowAsContextSource(): void {
+        this.data.getRows().forEach((row) => row.isContextMenuSource = this.selectedRowId === row.id ? true : false);
```

Comment:
can be a bit shorter: 
```ts
this.data.getRows().forEach((row) => row.isContextMenuSource = this.selectedRowId === row.id);
```

Or maybe even better: 
```ts
this.data.getRows().find((row) => this.selectedRowId = row.id)?.isContextMenuSource = true;
```

Second is even better because it does not iterate over all rows. It is just looking for first row which match condition and then loop stopped

Reply:
or even better if you use find: 
```ts
this.data.getRows().find((row) => this.selectedRowId === row.id)?.isContextMenuSource = true;
```

because then it is just looking for first row which meets condition during iteration and stopped loop.

Reply:
Indeed, it makes sense, but we need to mark 'isContextMenuSource' for other rows as false, as well. The test case that you have written check that for rest of the rows 'isContextMenuSource' must be false.

Reply:
I guess you set it to true because rows are reloaded right? So is not isContextMenuSource just undefined in that case? If yes then it is fine, it does not need to be false. It is js so undefined is falsy similar to false. 

So what is value of isContextMenuSource? If it is undefined then you don't need to set it to false because it is falsy too, unless it is true then yes you need set it to false.

Reply:
Okay, in that case, 'this.data.getRows().find((row) => this.selectedRowId === row.id)?.isContextMenuSource = true;' should work. Thanks for the comment.